server/handler/base.go

```diff
@@ -180,18 +191,39 @@ func (b *Base) EvaluateFetchedConfig(ctx context.Context, prctx pull.Context, cl
 			Str("event_trigger", trigger.String()).
 			Str("policy_trigger", policyTrigger.String()).
 			Msg("No evaluation necessary for this trigger, skipping")
-		return nil
+		return nil, nil
 	}
 
+	return evaluator, nil
+}
+
+func (b *Base) EvaluateFetchedConfig(ctx context.Context, prctx pull.Context, client *github.Client, evaluator common.Evaluator, fetchedConfig FetchedConfig) (common.Result, error) {
+	logger := zerolog.Ctx(ctx)
+
 	result := evaluator.Evaluate(ctx, prctx)
 	if result.Error != nil {
 		statusMessage := fmt.Sprintf("Error evaluating policy defined by %s", fetchedConfig)
 		logger.Warn().Err(result.Error).Msg(statusMessage)
-		err := b.PostStatus(ctx, prctx, client, "error", statusMessage)
-		return err
+
+		b.PostStatus(ctx, prctx, client, "error", statusMessage)
+
+		return result, result.Error
+	}
+
+	if err := b.PostStatusForResult(ctx, prctx, client, result); err != nil {
+		return result, err
+	}
+
+	if err := b.RequestReviewsForResult(ctx, prctx, client, result); err != nil {
```

Comment:
I'm not sure we want to request reviewers when loading the details page. Unlike posting statuses, review request have a side-effects and the design relies to some extent on limiting the number of cases when the code runs.

There's also a random delay in the code when requesting reviewers to try to space out events. Since events are asynchronous this is fine, but in the context of the details page, it will add latency.

Reply:
Sounds good. I was a little unsure about adding this to details rendering, so the context here is helpful. Perhaps we want to add something to that UI note about how to trigger reviewer requests on failed builds? (i.e. add a comment or commit to the PR)

Reply:
Let's leave it as-is and see if there's any user confusion should Policy Bot fail to request reviews. For instance, people might just find an appropriate chat room and ask for a review or tag some reviewers directly on the PR.